pkg/operator/ceph/cluster/cleanup.go

```diff
@@ -0,0 +1,180 @@
+you may not use this file except in compliance with the License.
+You may obtain a copy of the License at
+
+	http://www.apache.org/licenses/LICENSE-2.0
+
+Unless required by applicable law or agreed to in writing, software
+distributed under the License is distributed on an "AS IS" BASIS,
+WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+See the License for the specific language governing permissions and
+limitations under the License.
+*/
+
+// Package cluster to manage a Ceph cluster.
+package cluster
+
+import (
+	"fmt"
+	"time"
+
+	"github.com/pkg/errors"
+	cephv1 "github.com/rook/rook/pkg/apis/ceph.rook.io/v1"
+	"github.com/rook/rook/pkg/operator/ceph/cluster/mgr"
+	"github.com/rook/rook/pkg/operator/ceph/cluster/mon"
+	"github.com/rook/rook/pkg/operator/ceph/cluster/osd"
+	"github.com/rook/rook/pkg/operator/ceph/cluster/rbd"
+	"github.com/rook/rook/pkg/operator/ceph/controller"
+	"github.com/rook/rook/pkg/operator/ceph/file/mds"
+	"github.com/rook/rook/pkg/operator/ceph/object"
+	"github.com/rook/rook/pkg/operator/k8sutil"
+	"github.com/rook/rook/pkg/util"
+	batch "k8s.io/api/batch/v1"
+	v1 "k8s.io/api/core/v1"
+	metav1 "k8s.io/apimachinery/pkg/apis/meta/v1"
+)
+
+const (
+	clusterCleanUpPolicyRetryInterval = 5 //seconds
+)
+
+var (
+	volumnName      = "cleanup-volume"
+	appName         = "rook-ceph-cleanup"
+	dataDirHostPath = "ROOK_DATA_DIR_HOST_PATH"
+	namespaceDir    = "ROOK_NAMESPACE_DIR"
+)
+
+func (c *ClusterController) startClusterCleanUp(cluster *cephv1.CephCluster, cephHosts []string) {
+	logger.Infof("starting clean up for cluster %q", cluster.Name)
+	err := c.waitForCephDaemonCleanUp(cluster, time.Duration(clusterCleanUpPolicyRetryInterval)*time.Second)
+	if err != nil {
+		logger.Errorf("failed to wait till ceph daemons are destroyed. %v", err)
+		return
+	}
+
+	c.startCleanUpJobs(cluster, cephHosts)
+}
+
+func (c *ClusterController) startCleanUpJobs(cluster *cephv1.CephCluster, cephHosts []string) {
+	for _, hostName := range cephHosts {
+		logger.Infof("starting clean up job on node %q", hostName)
+		jobName := k8sutil.TruncateNodeName("cluster-cleanup-job-%s", hostName)
+		podSpec := c.cleanUpJobTemplateSpec(cluster)
+		podSpec.Spec.NodeSelector = map[string]string{v1.LabelHostname: hostName}
+		job := &batch.Job{
+			ObjectMeta: metav1.ObjectMeta{
+				Name:      jobName,
+				Namespace: cluster.Namespace,
+				Labels:    controller.AppLabels(appName, cluster.Namespace),
+			},
+			Spec: batch.JobSpec{
+				Template: podSpec,
+			},
+		}
+
+		if err := k8sutil.RunReplaceableJob(c.context.Clientset, job, true); err != nil {
+			logger.Error("failed to run cluster clean up job on node %q. %+v", hostName, err)
+		}
+	}
+}
+
+func (c *ClusterController) cleanUpJobContainer(cluster *cephv1.CephCluster) v1.Container {
+	volumeMounts := []v1.VolumeMount{}
+	hostPathVolumeMount := v1.VolumeMount{Name: volumnName, MountPath: cluster.Spec.DataDirHostPath}
```

Comment:
In the future I anticipate we won't require the host path. In this case only devices might be cleaned up. For now, I'd suggest just not adding the volume if the DataDirHostPath is empty. Also, if the cleanup policy doesn't request to cleanup the host path, we shouldn't mount it.

Reply:
adding `volumMounts` and  `envVars` related to hostpath only  cluster.Spec.CleanupPolicy.DeleteDataDirOnHosts != "" and  cluster.Spec.DataDirHostPath != ""